### ISSUE TITLE
Can i change channel count in tf.keras.preprocessing.image.ImageDataGenerator or tf.keras.preprocessing.image.DirectoryIterator?

### ISSUE THREAD

**Issue by DJLee68** [tensorflow/tensorflow#40698]
In Xception Pretrained model, i'm trying to use 6 channel as an input(299, 299, 6).

I'm using ImageDataGenerator and DirectoryIterator to load and preprocess data like this.

```
 image_data_generator_train = ImageDataGenerator(
                preprocessing_function = myFunc        
                validation_split=val_ratio,
    )
```

```
train_data = DirectoryIterator(
                directory=imagenet_train,
                image_data_generator=image_data_generator_train,          
                target_size=input_shape,
                classes=train_classes,
                class_mode='sparse',
                batch_size=train_batch_size,
                subset = 'training',
                shuffle=True,
                seed=2030,
                interpolation='bilinear'
)
```

and in myFunc, i'm trying to return 6 channel(299, 299, 6) but it occurs error because preprocessing_function's input and output should be same shape.
so i want to make input 6 channels, like concatenate input with channel like 
tf.concat([img, img], axis = 2) to make shape (299, 299,6)
can i do this in directoryiterator or imagedatagenerator?

**Comment by Saduf2019** [comment]
@DJLee68 
Please share complete indented code such that we could replicate the error faced, can you share the error or if possible share a colab gist with the error for us to analyse.
